symplyphysics/core/fields/operators.py

```diff
@@ -2,10 +2,46 @@
 
 from ..dimensions import ScalarValue
 from ..fields.vector_field import VectorField
+from ..fields.scalar_field import ScalarField
 from ..coordinate_systems.coordinate_systems import CoordinateSystem
 from ..vectors.vectors import Vector
 
 
+def gradient_operator(field: ScalarField) -> VectorField:
+    field_space = field.apply_to_basis()
+    if field.coordinate_system.coord_system_type == CoordinateSystem.System.CARTESIAN:
+        x = field.coordinate_system.coord_system.base_scalars()[0]
+        y = field.coordinate_system.coord_system.base_scalars()[1]
+        z = field.coordinate_system.coord_system.base_scalars()[2]
+        gradient = Vector([
+            diff(field_space, x),
+            diff(field_space, y),
+            diff(field_space, z)
+        ], field.coordinate_system)
+        return VectorField.from_vector(gradient)
+    if field.coordinate_system.coord_system_type == CoordinateSystem.System.CYLINDRICAL:
+        r = field.coordinate_system.coord_system.base_scalars()[0]
+        theta = field.coordinate_system.coord_system.base_scalars()[1]
+        z = field.coordinate_system.coord_system.base_scalars()[2]
+        gradient = Vector([
+            diff(field_space, r),
+            diff(field_space, theta) / r,
+            diff(field_space, z)
+        ], field.coordinate_system)
+        return VectorField.from_vector(gradient)
+    if field.coordinate_system.coord_system_type == CoordinateSystem.System.SPHERICAL:
+        r = field.coordinate_system.coord_system.base_scalars()[0]
+        theta = field.coordinate_system.coord_system.base_scalars()[1]
+        phi = field.coordinate_system.coord_system.base_scalars()[2]
+        gradient = Vector([
+            diff(field_space, r),
+            diff(field_space, theta) / r,
+            diff(field_space, phi) / (r * sin(theta))
```

Comment:
This is wrong. Azimuthal and polar angles have different notation here than in some books.

Reply:
I now understand that in your notation the angle names are swapped. However, it most literature and scientific journals  [like sciencedirect](https://www.sciencedirect.com/topics/mathematics/spherical-polar-coordinate) other notation is accepted, namely theta is the polar angle and phi is the azimuthal one. Maybe it is better to stick to more a more common convention so as not to make confusion?

Reply:
You're right but it is a major refactor and out of the scope of this PR